snowexsql/conversions.py

```diff
@@ -54,8 +53,10 @@ def query_to_geopandas(query, engine, **kwargs):
     # Fill out the variables in the query
     sql = query.statement.compile(dialect=postgresql.dialect())
 
-    # Get dataframe from geopandas using the query and engine
-    df = gpd.GeoDataFrame.from_postgis(sql, engine, **kwargs)
+    # Get dataframe from geopandas using the query and the DB connection.
+    # By passing in the actual connection, we maintain ownership of it and
+    # keep it alive until we close it.
+    df = gpd.read_postgis(sql, engine.connect(), **kwargs)
```

Comment:
This change to pass in the connection was key so we maintain ownership and pandas/geopandas won't close this on us.

Reply:
Using a `with` block would close the connection for us again and require another monkeypatch for the test suite. Let's do that in a follow up PR